modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js

```diff
@@ -17,244 +17,302 @@
+            var swatch = '<span class="detail_swatch" style="background-color: ' +
+                         series.color + '"></span>';
+            return swatch + series.name + ": " + parseInt(y) + '<br>' + date;
         }
+    });
 
-        function getData() {
-            if (deviceIndex >= devices.length) {
-                return;
-            }
-            var backendApiUrl = $("#connectedcup-div-chart").data("backend-api-url") + devices[deviceIndex].deviceIdentifier
-                                + "/sensors/" + type + "?from=" + from + "&to=" + to;
-            var successCallback = function (data) {
-                if (data) {
-                    drawLineGraph(JSON.parse(data));
-                }
-                deviceIndex++;
-                getData();
-            };
-            invokerUtil.get(backendApiUrl, successCallback, function (message) {
-                console.log(message);
-                deviceIndex++;
-                getData();
-            });
+    var hoverDetailCoffeelevel = new Rickshaw.Graph.HoverDetail({
+        graph: coffeelevelGraph,
+        formatter: function (series, x, y) {
+            var date = '<span class="date">' +
+                moment((x + tzOffset) * 1000).format('Do MMM YYYY h:mm:ss a') + '</span>';
+            var swatch = '<span class="detail_swatch" style="background-color: ' +
+                series.color + '"></span>';
+            return swatch + series.name + ": " + parseInt(y) + '<br>' + date;
         }
+    });
+
+    var shelving = new Rickshaw.Graph.Behavior.Series.Toggle({
+        graph: temperatureGraph,
+        legend: legend
+    });
 
-        function drawLineGraph(data) {
-            if (data.length === 0 || data.length === undefined) {
-                return;
+    var order = new Rickshaw.Graph.Behavior.Series.Order({
+        graph: temperatureGraph,
+        legend: legend
+    });
+
+    var highlighter = new Rickshaw.Graph.Behavior.Series.Highlight({
+        graph: temperatureGraph,
+        legend: legend
+    });
+
+    var shelvingCoffee = new Rickshaw.Graph.Behavior.Series.Toggle({
+        graph: coffeelevelGraph,
+        legend: legendCoffee
+    });
+
+    var orderCoffee = new Rickshaw.Graph.Behavior.Series.Order({
+        graph: coffeelevelGraph,
+        legend: legendCoffee
+    });
+
+    var highlighterCoffee = new Rickshaw.Graph.Behavior.Series.Highlight({
+        graph: coffeelevelGraph,
+        legend: legendCoffee
+    });
+
+    var deviceIndex = 0;
+
+    if (devices) {
+        getData();
+    } else {
+        var backendApiUrl = $("#connectedcup-div-chart").data("backend-api-url") + "/sensors/temperature" + "?from=" + from + "&to=" + to;
+        var successCallback = function (data) {
+            if (data) {
+                drawTemperatureLineGraph(JSON.parse(data));
             }
+        };
+        invokerUtil.get(backendApiUrl, successCallback, function (message) {
+            console.log(message);
```

Comment:
Replace console logging as appropriate. Stick to jaggery log module for any logging related need.